rust/lancedb/src/error.rs

```diff
@@ -46,8 +46,37 @@ pub enum Error {
     ObjectStore { source: object_store::Error },
     #[snafu(display("lance error: {source}"))]
     Lance { source: lance::Error },
-    #[snafu(display("Http error: {message}"))]
-    Http { message: String },
+    #[cfg(feature = "remote")]
+    #[snafu(display("Http error: (request_id={request_id}) {source}"))]
```

Comment:
How does snafu actually stringify `source`?  I'm wondering if there is value in forcing it to include the debug representation (reqwest error `to_string`'s are sometimes incomplete IIRC)

Reply:
I think ideally we are printing the full chain of errors. (Going through the source fields.) Though snafu doesn't seem to be the best for this. There's a good discussion about this here that I want to look through more deeply: https://stackoverflow.com/questions/62869360/should-an-error-with-a-source-include-that-source-in-the-display-output